src/backend/marsha/core/tests/test_xapi.py

```diff
@@ -0,0 +1,118 @@
+"""Tests for the xapi module of the Marsha project."""
+from unittest import mock
+
+from django.test import TestCase
+
+from ..exceptions import MissingUserIdError
+from ..factories import VideoFactory
+from ..xapi import XAPI, requests
+
+
+class MockLtiUser:
+    """Mock LTIUser class."""
+
+    @property
+    def user_id(self):
+        """Force to raise a wanted exception."""
+        raise AttributeError("user_id")
+
+
+class XAPITest(TestCase):
+    """Test the xapi module."""
+
+    def test_xapi_missing_user_id(self):
+        """Missing lti user_id should raise an exception."""
+        xapi = XAPI("https://lrs.example.com", "auth_token")
+        video = VideoFactory()
+
+        with self.assertRaises(MissingUserIdError):
+            xapi.send(video, {}, MockLtiUser())
+
+    @mock.patch.object(requests, "post")
+    def test_xapi_enrich_and_send_statement(self, mock_requests_post):
+        """XAPI statement sent by the front application should be enrich.
+
+        Before sending a statement, the xapi module is reponsible to enrich it.
+        """
+        xapi = XAPI("https://lrs.example.com", "auth_token")
+        video = VideoFactory(
+            id="68333c45-4b8c-4018-a195-5d5e1706b838",
+            playlist__consumer_site__domain="example.com",
+        )
+
+        mock_lti_user = mock.MagicMock()
```

Comment:
not sure if you really need to mock this since you are passing it yourself to the function below... ? Maybe just instantiate a LTIUser normally with what you want?

Reply:
I will have to mock the `TokenUser` isn't it ?